### PR TITLE
Detect Scala CLI project

### PR DESCRIPTION
**Pull request by wleczny**
This PR introduces `ScalaCliBuildTool` and detects project as Scala CLI project when `project.scala` file or `.scala-build` directory is present.

One of the requirements of https://github.com/VirtusLab/scala-cli/issues/1489

### CODE REVIEW

**Review by tgodzik — COMMENTED** [review]
Looks good! Could you just add the test, might be similar to `SbtServerSuite`

**Comment by wleczny** [comment]
Of course!

File: metals/src/main/scala/scala/meta/internal/metals/MetalsLspService.scala [hunk 1]
```diff
@@ -2349,14 +2350,25 @@ class MetalsLspService(
...
           }
-        case None =>
+        case Some(buildTool: BuildServerProvider) =>
+          buildTool.digest(workspace) match {
+            case None =>
+              scribe.warn(s"Skipping generating bsp files, no checksum.")
+              Future.successful(BuildChange.None)
+            case Some(digest) if !buildTools.isBsp =>
+              generateBspConfig()
+              indexer.reloadWorkspaceAndIndex(forceImport, buildTool, digest)
```
Comment (kasiaMarek) [inline metals/src/main/scala/scala/meta/internal/metals/MetalsLspService.scala:2360]: `reloadWorkspaceAndIndex` takes 4 args now and make sure to run `sbt scalafixAll` :)

**Comment by tgodzik** [comment]
Suceeded by https://github.com/scalameta/metals/pull/5383

Thanks for the hard work @wleczny !

**Pull request closed without merging**